### PR TITLE
Custom php interpreter support

### PR DESCRIPTION
**Pull request by timneutkens**
Attempt at providing support for different php interpreters.
Uses a subshell at the moment. I'm not quite certain that that is the way to go in bash. But if you want to use wp-cli like: /usr/bin/php wp <wp-cli command>. you have to specify a file path like: /usr/bin/php /path/to/wp <wp-cli command>. Otherwise it will throw a no input file specified error.

### CODE REVIEW

**Comment by westonruter** [comment]
What about adding something like a `wpcli_binary_path` config instead of using `$(command -v wp)`? I suppose the latter would still be needed if `wpcli_binary_path` is empty while `php_interpreter` is not empty.

**Comment by timneutkens** [comment]
Did a rework on this one. Combined your thoughts on it with mine. Added a new options called wpcli_binary_path in which you can specify the wp-cli command to be executed. So if you want to use a custom php interpreter it is possible. You also don't have the clutter of 2 options in the code.
I tweaked the original bash script a bit. It is now only used to detect or install wp-cli if no wpcli_binary_path is set.

File: wp-cli.sample.yml [hunk 1]
```diff
@@ -15,8 +15,8 @@ ssh:
     # We cd to this path on the remote server before running WP-CLI
     path: /var/www/staging.example.com/current/docroot
 
-    # Use a custom php interpreter on the remote server
-    php_interpreter: php
+    # Use a custom php interpreter / wp-cli command on the remote server
+    wpcli_binary_path: php ./vendor/bin/wp
```
Comment (westonruter) [inline wp-cli.sample.yml]: Instead of `wpcli_binary_path` then perhaps it should just be `wpcli_command`.

File: wp-cli.sample.yml [hunk 2]
```diff
@@ -15,6 +15,9 @@ ssh:
     # We cd to this path on the remote server before running WP-CLI
     path: /var/www/staging.example.com/current/docroot
 
+    # Use a custom php interpreter / wp-cli command on the remote server
+    wpcli_binary_path: php ./vendor/bin/wp
```
Comment (westonruter) [inline wp-cli.sample.yml]: Instead of `wpcli_binary_path` then perhaps it should just be `wpcli_command`.

**Comment by timneutkens** [comment]
I renamed wpcli_binary_path to wpcli_command.

File: wp-cli-ssh.php [hunk 3]
```diff
@@ -131,27 +131,32 @@ public function __invoke( $args, $assoc_args ) {
...
-			set -e;
+		$cmd = 'set -e;';
+		if( isset( $ssh_config['wpcli_command'] ) ) {
+			$cmd .= 'wp_command=' . escapeshellcmd( $ssh_config['wpcli_command'] ) . ';';
+		} else {
+			// Inline bash script to detect or download wp-cli
+			$cmd .= '
 			if command -v wp >/dev/null 2>&1; then
-				wp_command=wp;
+				wp_command=$(command -v wp);
```
Comment (westonruter) [inline wp-cli-ssh.php]: Is `$(command -v wp)` now doing anything meaningful here?

Comment (timneutkens) [inline wp-cli-ssh.php]: Good catch. Think this can be reverted to `wp` now.

Comment (timneutkens) [inline wp-cli-ssh.php]: Reverted it.

**Comment by westonruter** [comment]
👍 good work

**Comment by timneutkens** [comment]
@westonruter Thanks! Been needing this to execute the following gist:
https://gist.github.com/timneutkens/32c9bcd7d8576e663958ed4b1389c1dc

**Comment by westonruter** [comment]
@timneutkens please see conversation at https://github.com/wp-cli/wp-cli/pull/2755

**Comment by westonruter** [comment]
Specifically at https://github.com/wp-cli/wp-cli/pull/2755#discussion_r63400131

**Comment by timneutkens** [comment]
Will take a look tomorrow. Thanks for Referencing!

**Pull request merged**